### PR TITLE
Soft Fork Constraint Machine

### PR DESCRIPTION
**Pull request by talekhinezh**
* Implement ability to add forks on epoch boundaries with different constraint machine versions

### CODE REVIEW

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/CommittedReader.java [hunk 1]
```diff
@@ -30,4 +30,23 @@
 	VerifiedTxnsAndProof getNextCommittedTxns(DtoLedgerProof start);
 	Optional<LedgerProof> getEpochProof(long epoch);
 	Optional<LedgerProof> getLastProof();
+
+	static CommittedReader mocked() {
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/CommittedReader.java:34]: not a big fan of leaking test classes to `src/main`

**Review by LukasGasior1 — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/CommittedReader.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/sync/CommittedReader.java:34]: yeah this is quite a hack but it's required for GenerateUniverses, not just test classes

**Pull request merged**